### PR TITLE
refactor: SetItemMetadataField parameter struct

### PR DESCRIPTION
**Pull request by andreasgangso**

### CODE REVIEW

File: cmd/trigger_ui/main.go [hunk 1]
```diff
@@ -207,18 +207,22 @@ func (s *TriggerServer) vxExportPOST(ctx *gin.Context) {
...
-				err = s.vidispine.SetItemMetadataField(vxID, "", vscommon.FieldLangsToExport.Value, element)
-			} else {
-				err = s.vidispine.AddToItemMetadataField(vxID, "", vscommon.FieldLangsToExport.Value, element)
-
-			}
+			err = s.vidispine.SetItemMetadataField(vsapi.SetItemMetadataFieldParams{
+				ItemID: vxID,
+				Key:    vscommon.FieldLangsToExport.Value,
+				Value:  element,
+				Add:    i != 0,
```
Comment (KillerX) [inline cmd/trigger_ui/main.go:224]: This was intentionally left as two functions. It's the implicit vs. explicit thing.
The behavior is significantly different that IMO it should not be controlled by just an optional parameter.

Even worse, the implicit default is false, i.e. "replace".
You are one forgotten `Add: true` away from nuking a metadata field in an automated fashion.

File: services/vidispine/vsapi/chapter.go [hunk 2]
```diff
@@ -30,6 +25,7 @@ func (c *Client) GetChapterMeta(itemVXID string, inTc, outTc float64) (map[strin
 			continue
 		}
 
+		// TODO: @KillerX can you document why we need to do this? e.g. by extracting into a function
 		for _, field := range clip.Terse {
```
Comment (KillerX) [inline services/vidispine/vsapi/chapter.go:29]: Essentially we filter out chapters that are not part of the subclip:

```
+-------------------------------------------------------------------------------------------+
|                         Full Asset                                                        |
+-------------------------------------------------------------------------------------------+
+-----------------------------+                                                              
|      Chapter 1              |                                                              
+-----------------------------+                                                              
                   +--------------------------------+                                        
                   |       Chapter 2                |                                        
                   +--------------------------------+                                        
                                       +-------------------------------------+               
                                       |           Chapter 3                 |               
                                       +-------------------------------------+               
                             +--------------------------------------------------------------+
                             |                Chapter 4                                     |
                             +--------------------------------------------------------------+
                                                                                             
                                  +---------------------------------+                        
                                  |        CLIP                     |                        
                                  +---------------------------------+                        
```

In this case you only want Chapter 2-4 and not chapter 1.

In addition the chapter start and end times get cut to not extend beyond the clip start and end time.

**Review by KillerX — CHANGES_REQUESTED** [review]
I still think the `Add` and `Set` functions should be distinct as commented.

File: services/vidispine/vsapi/chapter.go (hunk 2 above)
Comment (andreasgangso) [reply services/vidispine/vsapi/chapter.go:28]: moved it to a func "trimTimecodesToBeWithinRange"

File: cmd/trigger_ui/main.go (hunk 1 above)
Comment (andreasgangso) [reply cmd/trigger_ui/main.go]: split it back up, so this PR is basically just about the params and cleanup now

**Review by KillerX — APPROVED** [review]

**Pull request merged**